IPs/RSKIP419.md

```diff
@@ -0,0 +1,229 @@
+
+
+### Proof transaction creation
+
+Once the funding transaction is registered, the Proof will be created for the proposed Powpeg to sign and broadcast it so it can prove it can spend funds.
+
+##### Funding transaction registration
+The funding transaction should be registered through the registerBtcTransaction method.
+For that purpose, the method will be modified to be able to identify it.
+It should verify:
+- SVP period is ongoing
+(i.e. proposedFederation exists & CURRENT_RSK_BLOCK < PROPOSED_FED_CREATION_BLOCK_NUMBER + SVP_PERIOD)
+- Transaction hash without signatures matches the record in the FUND_TX_HASH_UNSIGNED storage entry
+
+Once everything is verified, the transaction will be registered and the transaction hash will be saved in a new storage entry, FUND_TX_HASH_SIGNED.
+The funding transaction hash signed being saved means that the proof Transaction can be created.
+
+**Important remark**: This method will be public, meaning any user could call it.
+
+#### Proof transaction creation or validation process failure
+
+A new method, `processPowpegChangeValidation` will be called from the `updateCollections` method.
+
+The `processPowpegChangeValidation` will perform some actions if a proposedFederation exists depending on this two scenarios:
+- SVP period ended. This implies the validation was not successful.
+    -  The Bridge will emit a new event,
+        
+        `COMMIT_FEDERATION_FAILED(blockNumber indexed uint)`
+    - The proposed Powpeg will be eliminated.
+    - The election will be allowed once again.
+
+- SVP period didn't end        
+    - If the funding transaction signed hash is registered in the FUND_TX_HASH_SIGNED storage entry and the Proof transaction is not created yet, the Bridge will:
+        - Create it as described [here](#proof-transaction)
+        - Add it to a new storage set, key `spendTxWaitingForSignatures`
+        - Set it in a new Storage entry:
+            ```
+            KEY: "SPEND_TX_HASH"
+            VALUE: hash of SVP proof tx
+            ```    
+
+##### Proof transaction
+
+The Proof transaction is a Bitcoin transaction where an UTXO from the proposed Powpeg is sent to the current Powpeg.
+This transaction will be as simple and small as possible to avoid spending much satoshis as fees.
+
+Proof transaction structure:
+```
+{
+    inputs: [
+        {
+            txid: FUND_TX_HASH_SIGNED,
+            output_index: FUNDING_TX_OUTPUT_INDEX
+        }
+    ],
+    outputs: [
+        {
+            script: CURRENT_POWPEG_OUTPUT_SCRIPT,
+            value: TOTAL_UTXO - FEES
+        }
+    ]
+}
+```
+
+#### Proof transaction confirmation
+
+The SVP proof transaction should go through a new "peg-out signing process", since it has to be signed for the proposed pegnatories which belong neither the active nor the retiring powpeg, but the proposed one.
+
+#### Proof transaction signing
```

Comment:
How is the powpeg gonna get the information of the proof transaction waiting for signatures? Should we update `getStateForRelease` method or it should it be a new method?

Reply:
i would say a new method is better

Reply:
but thats a hard part we have to discuss deeply

Reply:
sounds good!